caikit/runtime/http_server/http_server.py

```diff
@@ -534,12 +551,18 @@ async def _generator() -> pydantic_response:
                             inference_func_name=model.get_inference_signature(
                                 output_streaming=True, input_streaming=False
                             ).method_name,
+                            aborter=aborter,
                             **request_params,
                         )
                     ):
                         yield ServerSentEvent(
                             data=result.to_json(), event=StreamEventTypes.MESSAGE.value
                         )
+
+                    # Clean up Aborter to ensure object is deleted
+                    if aborter:
+                        aborter.abort()
```

Comment:
This smells like it should be a context manager?

```
with HttpRequestAborter(context) as aborter:
    ...
    self.global_predict_servicer.predict_model(
        ...
        aborter=aborter
    )
```
(I'm just guessing there's a case where a non-aborting-related exception could cause this cleanup to be skipped)

If constructing one of these is light, we can probably construct it always since the `AbortableAction` won't be created later if `config.runtime.use_abortable_threads` is false

Reply:
That's a good question. I'm not sure how lightweight it is since it runs an async task in the background.  However, I think this might be a perfect usecase for [nullcontext!](https://docs.python.org/3.7/library/contextlib.html#contextlib.nullcontext) (I've never had the chance to use one for real!). I'll push a commit with what I mean